en/docs/how-tos/passing-enduser-attributes-to-the-backend-using-jwt.md

```diff
@@ -175,6 +175,69 @@ Outer `[jwtGeneratorConfig]` configuration contains the properties and the custo
+`<WSO2 Key Manager URL>/keymanager-operations/user-info/claims/generate` endpoint. If you are willing to use the same 
+implementation, you can add the following configuration to the `micro-gw.conf` file.
+
+```toml
+[jwtGeneratorConfig.claimRetrieval]
+      retrieverImpl="org.wso2.micro.gateway.jwt.generator.MGWClaimRetrieverImpl"
+      [jwtGeneratorConfig.claimRetrieval.configuration]
+          # The User can provide any key-value pair here. These are the properties used by default.
+          # serverUrl = <Key Manager URL>
+          # username = <APIM Credentials Username>
+          # password = <APIM Credentials Password>
+```
+
+The `retrieveImpl` Key is used to provide the claim retriever Implementation. Since the value given here denotes an 
+inbuilt implementation, the user does not need to add any additional jar files to the project. Under the 
+`[jwtGeneratorConfig.claimRetrieval.configuration]`, the user can customize the properties as required. 
+
+In addition to that, it is required to enable claimRetrieval in `[[jwtTokenConfig]]` (per JWT issuer) and 
+`[keyManager]` Configurations based on user requirement. 
+
+``` toml tab="jwtTokenConfig"
+[[jwtTokenConfig]]
+    remoteUserClaimRetrievalEnabled = true
+```
+
+``` toml tab="keyManager"
+[keyManager]
+    remoteUserClaimRetrievalEnabled = true
+```
+
+!!! note
+    If you do not declare the `jwtGeneratorConfig.claimRetrieval.retrieverImpl` property in `micro-gw.conf`, 
+    the classloading operation will not happen within the microgateway. Hence, the claim retrieval process will 
+    not be executed.
+        
+
+There can be usecases where the users need to come up with their own claim retrieval Implementation. In such 
+a scenario, the user needs to implement the specific class extended from 
+[AbstractMGWClaimRetriever](https://github.com/wso2/product-microgateway/blob/v3.2.0/components/micro-gateway-jwt-generator/src/main/java/org/wso2/micro/gateway/jwt/generator/AbstractMGWClaimRetriever.java) class.
+For that, the user can have the same maven dependency used for custom JWT Generation implementation. A sample 
+implementation is available [here](https://github.com/wso2/product-microgateway/tree/master/samples/sample-jwt-generator).
+In the end, you can add the corresponding jar file to the microgateway project's `/lib` directory prior to executing 
+`micro-gw build` command.
+
+```xml tab="Maven dependency"
+<dependencies>
+    <dependency>
+        <groupId>org.wso2.am.microgw</groupId>
+        <artifactId>mgw-jwt-generator</artifactId>
+        <version>3.2.0</version>
+    </dependency>
+</dependencies>
+```
+
+!!! note
+    Please note that if you have used any additional dependencies to the customized implementation, you may need to add all
+    of them to the microgateway project's `/lib` directory. This is because if the microgateway can't find 
+    those dependencies within the project, the functionality will not be available.
+
```

Comment:
section #### "Setting the header of the backend request" is currently under "JWT Generator Configurations". Is it correct?

Reply:
updated via https://github.com/wso2/docs-mg/pull/43/commits/9480fc53f208b95446e6f3e220351d4b5229fa1c